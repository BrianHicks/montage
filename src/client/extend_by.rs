#[cynic::schema("montage")]
mod schema {}

#[derive(cynic::QueryVariables, Debug)]
pub struct ExtendByMutationVariables {
    pub duration: Duration,
}

#[derive(cynic::QueryFragment, Debug)]
#[cynic(graphql_type = "Mutation", variables = "ExtendByMutationVariables")]
pub struct ExtendByMutation {
    #[arguments(duration: $duration)]
    pub extend_by: Session,
}

#[derive(cynic::QueryFragment, Debug)]
pub struct Session {
    pub description: String,
    pub projected_end_time: DateTime,
}

type DateTime = chrono::DateTime<chrono::Local>;
cynic::impl_scalar!(DateTime, schema::DateTime);

type Duration = iso8601::Duration;
cynic::impl_scalar!(Duration, schema::Duration);

#[cfg(test)]
mod test {
    use super::*;
    use cynic::MutationBuilder;

    #[test]
    fn gql_output() {
        let operation = ExtendByMutation::build(ExtendByMutationVariables {
            duration: iso8601::duration("PT25M").unwrap(),
        });

        insta::assert_snapshot!(operation.query);
    }
}
