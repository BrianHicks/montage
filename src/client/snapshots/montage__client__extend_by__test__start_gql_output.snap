---
source: src/client/extend_by.rs
expression: operation.query
---
mutation ExtendByMutation($duration: Duration!) {
  extendBy(duration: $duration) {
    description
    projectedEndTime
  }
}


